### PR TITLE
Various Bash Improvements

### PR DESCRIPTION
**Pull request by Androbin**
Sorry about the previous spamming.
I didn't expect it to sum up that quickly.

### CODE REVIEW

**Comment by jhseu** [comment]
Thanks for merging!

Jenkins, test this please

**Comment by Androbin** [comment]
I have quickly repaired some changes.
Retry.

**Comment by gunan** [comment]
Jenkins, test this please.

File: tensorflow/tools/ci_build/ci_build.sh [hunk 1]
```diff
@@ -80,7 +80,7 @@ fi
 
 # cmake (CPU) builds do not require configuration.
 if [[ "${CONTAINER_TYPE}" == "cmake" ]]; then
-  CI_COMMAND_PREFIX=""
+  CI_COMMAND_PREFIX=("")
```
Comment (gunan) [inline tensorflow/tools/ci_build/ci_build.sh]: why do we need the parenthesis here?

Comment (Androbin) [inline tensorflow/tools/ci_build/ci_build.sh]: The variable was used as an array but is now assigned a string. Bash behaves a little weird here:
https://github.com/koalaman/shellcheck/wiki/SC2178

**Comment by Androbin** [comment]
I am sorry, the last few commits don't belong here.

File: tensorflow/tools/ci_build/ci_build.sh (hunk 1 above)
Comment (Androbin) [inline tensorflow/tools/ci_build/ci_build.sh]: This kind of unintended behavior is what this PR tries to fix.
There are so many things seriously wrong with Bash.
It is just way too easy to make mistakes.

File: tools/tf_env_collect.sh [hunk 2]
```diff
@@ -70,29 +72,31 @@ EOF
...
-  echo DYLD_LIBRARY_PATH ${DYLD_LIBRARY_PATH}  >> $OUTPUT_FILE;
-fi
-
-
-echo >> $OUTPUT_FILE >> $OUTPUT_FILE
-echo '== nvidia-smi ===================================================' >> $OUTPUT_FILE
-nvidia-smi 2>&1 >> $OUTPUT_FILE
-
-echo >> $OUTPUT_FILE
+grep libcudnn.so /tmp/loadedlibs
```
Comment (jhseu) [inline tools/tf_env_collect.sh]: This belongs in the block below, right?

Comment (jhseu) [inline tools/tf_env_collect.sh]: (I'll just make the edit.)

**Review by jhseu — APPROVED** [review]

**Comment by jhseu** [comment]
Jenkins, test this please

**Comment by Androbin** [comment]
I'd say, the failing `//tensorflow/tools/api/tests:api_compatibility_test` should be unrelated?

**Pull request merged**